adsApp/src/ADSPortDriver.cpp

```diff
@@ -80,7 +80,10 @@ DeviceVariable *ADSPortDriver::createDeviceVariable(DeviceVariable *baseInfo) {
 
     if (adsDeviceVar->adsPV->addr->get_operation() == Operation::Read) {
         ads_read_vars.push_back(adsDeviceVar->adsPV);
-    } else {
+    } else if (adsDeviceVar->adsPV->addr->get_operation() == Operation::Write) {
+        ads_write_vars.push_back(adsDeviceVar->adsPV);
+    } else if (adsDeviceVar->adsPV->addr->get_operation() == Operation::ReadWrite) {
+        ads_read_vars.push_back(adsDeviceVar->adsPV);
```

Comment:
Looking at things from the user's perspective, having to define both the `RW` operation and the `asyn:READBACK` info on the record seems quite redundant. I think it would be better if just one of these was sufficient, and it has to be `asyn:READBACK` because asyn needs to know about this, not just the driver.

Instead of relying on the `RW` flag, the driver can recognize the need for readback when a variable is both a "write" variable and an "interrupt" variable. The latter piece of information is not available in the `createDeviceVariable()` function because that one is called too early. But it should be available in the `initHook()`. I think going over the array returned by `getInterruptVariables()`, taking those with the write operation, and appending them to `ads_read_vars` before sum read buffers are allocated, should be enough.

Reply:
I wasn't aware that `asyn:READBACK` gets the variable in the `getInterruptVariables()` list. I thought `field(SCAN, "I/O Intr")` is needed. But you are right. I implemented it in `initHook()`, but I had to add a flag to the `ADSVariable` class, so I can handle the `await_written_readback` and the block of write variables in the read methods of `ADSPortDriver`. I hope that's OK.

And sorry for the additional merge commit. You should consider squashing this PR into one commit when merging.